### PR TITLE
add ability-collapse-expand-token

### PR DESCRIPTION
**Pull request by MightyDevKnight**

### CODE REVIEW

**Comment by six7** [comment]
Can you add test coverage for this new functionality?

File: src/app/components/TokenTree.tsx [hunk 1]
```diff
@@ -32,57 +65,93 @@ const TokenTree: React.FC<Props> = ({
...
               <div className="flex items-center justify-between group">
-                <TokenGroupHeading label={name} path={stringPath} id="listing" type={schema.type} />
+                <StyledCollapsableTokenGroupHeadingButton
+                  collapsed={collapsed.includes(item.stringPath)}
+                  data-cy={`tokenlisting-group-${item.stringPath}`}
+                  type="button"
+                  onClick={onToggleCollapsed}
+                >
+                  {collapsed.includes(item.stringPath) ? (
+                    <svg width="6" height="6" viewBox="0 0 6 6" xmlns="http://www.w3.org/2000/svg">
```
Comment (LiamMartens) [inline src/app/components/TokenTree.tsx:120]: Should these be moved to svg files?

**Review by LiamMartens — CHANGES_REQUESTED** [review]

File: src/app/components/TokenTree.test.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+const showNewForm = jest.fn();
+const showForm = jest.fn();
+
+describe('TokenTree', () => {
+  beforeEach(() => {
+    resetStore();
+  });
+
+  it('should be able to collpase token tree', async () => {
```
Comment (six7) [inline src/app/components/TokenTree.test.tsx:71]:
```suggestion
  it('should be able to collapse a token group', async () => {
```

File: src/app/components/TokenTree.test.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    />);
+
+    await fireEvent.click(getByTestId('tokenlisting-group-size.font'));
+    await fireEvent.click(getByTestId('tokenlisting-group-color'));
+
+    const { collapsedTokens } = store.getState().tokenState;
+    expect(collapsedTokens).toEqual(['size.font', 'color']);
+  });
+
+  it('should be able to expand token tree', async () => {
```
Comment (six7) [inline src/app/components/TokenTree.test.tsx:87]:
```suggestion
  it('should be able to expand a token group', async () => {
```

File: src/app/components/TokenTree.test.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    await fireEvent.click(getByTestId('tokenlisting-group-size.font'));
+    await fireEvent.click(getByTestId('tokenlisting-group-size.font'));
+    await fireEvent.click(getByTestId('tokenlisting-group-color'));
+
+    const { collapsedTokens } = store.getState().tokenState;
+    expect(collapsedTokens).toEqual(['color']);
+  });
+
+  it('should be able to work add a new token', async () => {
```
Comment (six7) [inline src/app/components/TokenTree.test.tsx:104]:
```suggestion
  it('should be able to add a new token', async () => {
```

File: src/app/components/TokenTree.tsx [hunk 5]
```diff
@@ -19,6 +23,36 @@ export type ShowNewFormOptions = {
...
 };
 
+const StyledTokenGroup = styled('div', {
+  display: 'flex',
+  flexDirection: 'row',
+  flexWrap: 'wrap',
+  justifyContent: 'flex-start',
+});
+
+const StyledCollapsableTokenGroupHeadingButton = styled('button', {
```
Comment (six7) [inline src/app/components/TokenTree.tsx:33]: can we move this to its own file? same with the StyledTokenGroup above? That way this file doesnt contain styling

**Review by six7 — CHANGES_REQUESTED** [review]
some minor comments, otherwise this is good!

**Review by six7 — APPROVED** [review]

**Pull request merged**